src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Since you have done the action of finding the gender you want, now you can further improve your code, in order not to call this function twice, you can first find people with the gender you want, save it to a variable, and use here as you see on this ?
And now it will be easier for you with the parameter names inside the array methods you use, and don't make the names so short